testing/trino-product-tests/src/main/java/io/trino/tests/product/utils/QueryExecutors.java

```diff
@@ -126,7 +126,7 @@ public static QueryExecutor onDelta()
 
         return new QueryExecutor()
         {
-            private final QueryExecutor delegate = testContext().getDependency(QueryExecutor.class, "delta");
+            private final QueryExecutor delegate = new DeltaQueryExecutor(testContext());
```

Comment:
`onDelta` creates new `DeltaQueryExecutor` every time it's called,
so caching inside `DeltaQueryExecutor` doesn't help much.
is `ArtifactResolver` indeed being called over and over again?

caching within `testContext`  wouldn't be awesome either, as that's created anew for every test or test class (idk exactly)

for class/classloader loading it would be better to have a static cache.